tools/bug-report/pkg/processlog/processlog.go

```diff
@@ -53,6 +60,54 @@ func Process(config *config.BugReportConfig, logStr string) (string, *Stats) {
 	return out, getStats(config, out)
 }
 
+func ProcessZtunnel(config *config.BugReportConfig, logStr string) (string, *Stats) {
+	var sb strings.Builder
+	stats := &Stats{}
+	start := config.StartTime
+	end := config.EndTime
+	for _, l := range strings.Split(logStr, "\n") {
+		t, level, line, valid := processZtunnelLogLine(l)
+		if len(level) != 0 {
+			switch strings.ToLower(level) {
+			case levelFatal:
+				stats.numFatals++
+			case levelError:
+				stats.numErrors++
+			case levelWarn:
+				stats.numWarnings++
+			}
+		}
+		if !valid {
+			// This maybe some configs or other logs, just append it.
+			sb.WriteString(line)
+			sb.WriteString("\n")
+			continue
+		}
+		if (t.Equal(start) || t.After(start)) && (t.Equal(end) || t.Before(end)) {
+			sb.WriteString(line)
+			sb.WriteString("\n")
+		}
+	}
+	return sb.String(), stats
+}
+
+func removeColor(s string) string {
+	return matchEscape.ReplaceAllString(s, "")
+}
+
+func processZtunnelLogLine(l string) (timestamp *time.Time, level string, text string, valid bool) {
+	line := removeColor(l)
```

Comment:
Can you open an issue to do this in ztunnel itself? to match Istio standard log format

Reply:
I've added a TODO here https://github.com/istio/istio/pull/43878/files#diff-b2eb21e6596c35d1bc5f18fec84e9e6ef18a57c8306897634ce489e5a7bc3692R312-R313, should we just use the plain ztunnel log or do the custom log processing like mine approach, and remove it once the issue is addressed?

Reply:
I thought we already removed the colors now?

The rest seems to be to fill in `importance` but that is never used,so may be a waste of effort?

Reply:
You right. However it's still tricky since ztunnel log format is separated with space instead of tab. Currently I see nowhere to set it in logging. I left with a new TODO https://github.com/istio/istio/pull/43878/files#diff-b2d1fb3eb896654dbad3c7c5845c065c08ed1684b8b3fe8837b18b1bf6328a5bR109.